pkg/apis/keycloak/v1alpha1/keycloak_types.go

```diff
@@ -4,26 +4,60 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // KeycloakSpec defines the desired state of Keycloak
 // +k8s:openapi-gen=true
 type KeycloakSpec struct {
-	AdminCredentialSecret string `json:"adminCredentialSecret,omitempty"`
+	// A list of extensions, where each one is a URL to a JAR files that will be deployed in Keycloak.
 	// +listType=set
-	Extensions       []string                 `json:"extensions,omitempty"`
-	Instances        int                      `json:"instances,omitempty"`
-	ExternalAccess   KeycloakExternalAccess   `json:"externalAccess,omitempty"`
+	// +optional
+	Extensions []string `json:"extensions,omitempty"`
+	// Number of Keycloak instances in HA mode. Default is 1.
+	// +optional
+	// +kubebuilder:default=1
+	Instances int `json:"instances,omitempty"`
+	// Controls external Ingress/Route settings.
+	// +optional
+	ExternalAccess KeycloakExternalAccess `json:"externalAccess,omitempty"`
+	// Controls external database settings.
+	// Using an external database requires providing a secret containing credentials
+	// as well as connection details. Here's an example of such secret:
+	//
+	//     apiVersion: v1
+	//     kind: Secret
+	//     metadata:
+	//         name: keycloak-db-secret
+	//         namespace: keycloak
+	//     data:
+	//         POSTGRES_DATABASE: cm9vdA==
```

Comment:
Would be nicer if example used string data so it's readable

Reply:
POSTGRES_EXTERNAL_PORT is this optional? If so perhaps mention it

Reply:
I've just checked - it's mandatory. Created [KEYCLOAK-12304](https://issues.jboss.org/browse/KEYCLOAK-12304).

Reply:
Fixed.